guides/eslint-migration.md

```diff
@@ -0,0 +1,311 @@
+2. **For each package:**
+   - Remove `.eslintrc*` files
+   - Add `eslint.config.ts` as above
+   - Remove local plugin deps
+   - Run lint and fix errors
+   - Run tests
+3. **Commit and open PR**
+
+---
+
+## Troubleshooting Guide
+
+### Common Issues and Solutions
+
+#### 1. **Jiti Version Compatibility Error**
+**Error:** `Error: You are using an outdated version of the 'jiti' library. Please update to the latest version of 'jiti' to ensure compatibility and access to the latest features.`
+
+**Solution:**
+- Add `jiti: "^2.4.2"` to the package's `devDependencies`
+- ESLint 9.x requires jiti >= 2.2.0, but monorepo dependencies might provide older versions
+
+#### 2. **TypeScript Project Service Errors**
+**Error:** `was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject`
+
+**Solutions:**
+- **Create/update `tsconfig.json`** that extends the base config:
+  ```json
+  {
+    "extends": "../../packages/ts/tsconfig.json",
+    "compilerOptions": {
+      "esModuleInterop": true,
+      "allowJs": true,
+      "checkJs": false
+    },
+    "include": [
+      "src/**/*",
+      "cypress/**/*",
+      "*.js",
+      "*.ts",
+      "*.jsx",
+      "*.tsx"
+    ],
+    "exclude": ["node_modules", "dist"]
+  }
+  ```
+- **For Cypress packages**, ensure `cypress/tsconfig.json` includes all test files:
+  ```json
+  {
+    "compilerOptions": {
+      "types": ["cypress"]
+    },
+    "include": [
+      "**/*.ts",
+      "**/*.js"
+    ]
+  }
+  ```
+- **Add `allowDefaultProject: true`** to ESLint config for problematic files:
+  ```ts
+  {
+    files: ['**/*.js', '**/*.ts', '**/*.jsx', '**/*.tsx'],
+    languageOptions: {
+      parserOptions: {
+        allowDefaultProject: true,
+      },
+    },
+  }
+  ```
+
+#### 3. **Skip Comment Rule Violations**
+**Error:** `@cypress/dev/skip-comment` rule violations for `it.skip()` tests
+
+**Solution:**
+- Replace `eslint-disable-next-line @cypress/dev/skip-comment` with proper explanatory comments:
+  ```js
+  // NOTE: This test is skipped for demonstration purposes
+  it.skip('first test', () => {})
+  ```
+
+#### 4. **Rule Mismatches in Pre-commit Hook**
+**Error:** ESLint rule violations that don't match the package's ESLint configuration (e.g., `Unexpected console statement` when `no-console` is off in package config)
+
+**Root Cause:** Pre-commit hook runs ESLint from root directory using root-level config, not package-specific config
+
+**Solution:**
+- Add package-specific lint-staged rule in root `package.json`:
+  ```json
+  "lint-staged": {
+    "npm/grep/**/*.{js,jsx,ts,tsx}": "cd npm/grep && eslint --fix",
+    "*.{js,jsx,ts,tsx,json,eslintrc,vue}": "eslint --fix"
+  }
+  ```
+
+#### 5. **ESLint Script Changes**
+**Before ESLint 9.x:**
+```json
+"lint": "eslint . --ext .js,.ts"
+```
+
+**After ESLint 9.x:**
+```json
+"lint": "eslint ."
```

Comment:
The `.` is also no longer needed. See https://eslint.org/docs/latest/use/command-line-interface